pkg/apis/componentconfig/types.go

```diff
@@ -24,6 +24,10 @@ type KubeProxyConfiguration struct {
 	// bindAddress is the IP address for the proxy server to serve on (set to 0.0.0.0
 	// for all interfaces)
 	BindAddress string `json:"bindAddress"`
+	// clusterCIDR is the CIDR range of the pods in the cluster. It is used to
+	// bridge traffic coming from outside of the cluster. If not provided,
+	// no off-cluster bridging will be performed.
+	ClusterCIDR string `json:"clusterCIDR"`
```

Comment:
I considered that feedback, but this is not the only place such a change
would have to go, so I thought that, for expedient merging, I would defer
and pay for it later.  It changes the design quite a bit due to the nature
of iptables rules.

If we want a merge today, I think we should forego it right now.
On Apr 19, 2016 7:09 AM, "Robert Bailey" <notifications@github.com> wrote:

> In pkg/apis/componentconfig/types.go
> <https://github.com/kubernetes/kubernetes/pull/24429#discussion_r60237383>
> :
>
> > @@ -24,6 +24,10 @@ type KubeProxyConfiguration struct {
> >  	// bindAddress is the IP address for the proxy server to serve on (set to 0.0.0.0
> >  	// for all interfaces)
> >  	BindAddress string `json:"bindAddress"`
> > +	// clusterCIDR is the CIDR range of the pods in the cluster. It is used to
> > +	// bridge traffic coming from outside of the cluster. If not provided,
> > +	// no off-cluster bridging will be performed.
> > +	ClusterCIDR string `json:"clusterCIDR"`
>
> To be clear, what I'm concerned about is that today we allocate a single
> consecutive /14 or /16 range for a cluster, which this code covers. But if
> we want a cluster to be able to have multiple /14s (e.g. for >1k nodes
> where a user doesn't have consecutive ip space available) then this is
> going to be fragile and we are either going to break compatibility (by
> changing this type) or introduce another variable (which is really ugly).
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24429/files/0dd7cf8e881c1ba816e22a5ce1808bc29ccbe1df#r60237383>
>